rockcraft/project.py

```diff
@@ -305,7 +319,7 @@ def _validate_all_platforms(cls, platforms):
 
     @pydantic.validator("parts", each_item=True)
     @classmethod
-    def _validate_parts(cls, item):
+    def _validate_parts(cls, item: Any) -> Any:
```

Comment:
It can help the type checker here to make a `TypeVar` so it knows we're returning the same type as we're  accepting, especially if this method becomes more complex in future.

Reply:
In this case I updated it to `Dict[str, Any]` because this `_validate_parts()` function just forwards the parameter to `rockcraft.parts.validate_part()` which has that annotation. Thanks